syntax = "proto3";

package datagen;

message DataGenRequest {
    int32 n_data = 1;
}

message DataGenResponse {
    bytes data = 1;
    string name = 2;
}

message MeasurePerformanceRequest {
}

message MeasurePerformanceResponse {
    float score = 1;
}

service DataGenService{
    rpc MeasurePerformance (MeasurePerformanceRequest) returns (MeasurePerformanceResponse) {}
    rpc DataGen (DataGenRequest) returns (DataGenResponse) {}
    rpc DataGenStream (DataGenRequest) returns (stream DataGenResponse) {}
}
